charts/controls/Controls.tsx

```diff
@@ -435,109 +443,81 @@ class TimelineControl extends React.Component<{ chart: ChartConfig }> {
         this.props.chart.useTimelineDomains = false
     }
 
-    render() {
-        // Todo: cleanup this method
-        const { chart } = this.props
-        if (chart.props.tab === "table") {
-            const { dataTableTransform } = chart
-            const years = dataTableTransform.timelineYears
-            if (years.length === 0) {
-                return null
-            }
+    @computed get startYear() {
+        const { activeTab, chart } = this.props
+        if (activeTab === "table")
+            return (
+                chart.dataTableTransform.autoSelectedStartYear ??
+                chart.timeDomain[0]
+            )
+        if (activeTab === "map") return chart.mapTransform.targetYearProp
+        else return chart.activeTransform.startYear!
+    }
 
-            const startYear =
-                dataTableTransform.autoSelectedStartYear ?? chart.timeDomain[0]
-            const endYear = chart.multiMetricTableMode
-                ? startYear
+    @computed get endYear() {
+        const { activeTab, chart } = this.props
+        if (activeTab === "table")
+            return chart.multiMetricTableMode
+                ? chart.dataTableTransform.startYear
                 : chart.timeDomain[1]
+        if (activeTab === "map") return chart.mapTransform.targetYearProp
+        else return chart.activeTransform.endYear!
+    }
 
+    componentDidUpdate(prevProps: TimelineControlProps) {
+        if (
+            prevProps.activeTab !== this.props.activeTab &&
+            this.props.activeTab != "map"
+        )
+            this.onChartTargetChange({
+                targetStartYear: this.startYear,
+                targetEndYear: this.endYear
+            })
+    }
+
+    @computed get timelineProps(): TimelineProps {
+        const { activeTab, chart } = this.props
+        return {
+            chart: chart,
```

Comment:
nit: just `chart` will do. I think there's a prettier rule or something we can add that will do this for us.

Reply:
prettier doesn't, but eslint does: [object-shorthand](https://eslint.org/docs/rules/object-shorthand)

Reply:
ooh, right!